### DISCUSSION TITLE
Query M2M

### DISCUSSION THREAD

**Discussion by tonxxd** [directus/directus#5852] [Archive]
Hello, I have a question regarding querying M2M relationships.

I have a collection e.g. Posts and a M2M gallery field. This creates a third collection posts_directus_files to store the relationships.

If I want to query the posts and populate the gallery field I have to use either REST 
api.com/items/posts?fields=gallery.directus_files_id.filename_disk 

or graphql 

```graphql
posts {
    gallery {
       directus_files_id: { 
            filename_disk
       }
    }
}
```

Is this the intended behavior? Is there a way to have something like the following?

```graphql
posts {
    gallery {
       filename_disk
    }
}
```
Thanks in advance

**Comment by rijkvanzanten** [comment]
No, a many to many is nothing more than a one-to-many + many-to-one. You can have any other fields on that junction table, so we don't strip it out by default 👍🏻

**Reply by tonxxd** [reply]
Makes sense, thanks!
